guvnor-ala/guvnor-ala-registry-vfs/src/main/java/org/guvnor/ala/registry/vfs/VFSRegistryHelper.java

```diff
@@ -0,0 +1,239 @@
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.guvnor.ala.marshalling.Marshaller;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.io.IOService;
+import org.uberfire.java.nio.IOException;
+import org.uberfire.java.nio.file.DirectoryStream;
+import org.uberfire.java.nio.file.FileSystem;
+import org.uberfire.java.nio.file.Path;
+
+/**
+ * Helper class for implementing the different VFS based registries that are backed by using VFSRegistryEntry elements.
+ */
+@ApplicationScoped
+public class VFSRegistryHelper {
+
+    protected static final String PROVISIONING_BRANCH = "master";
+
+    protected static final String PROVISIONING_PATH = "provisioning";
+
+    private static final Logger logger = LoggerFactory.getLogger(VFSRegistryHelper.class);
+
+    private VFSMarshallerRegistry marshallerRegistry;
+
+    private IOService ioService;
+
+    private FileSystem fileSystem;
+
+    private Path provisioningRootPath;
+
+    private VFSRegistryEntryMarshaller entryMarshaller;
+
+    public VFSRegistryHelper() {
+        //Empty constructor for Weld proxying
+    }
+
+    @Inject
+    public VFSRegistryHelper(final VFSMarshallerRegistry marshallerRegistry,
+                             final @Named("ioStrategy") IOService ioService,
+                             final @Named("systemFS") FileSystem fileSystem) {
+        this.marshallerRegistry = marshallerRegistry;
+        this.ioService = ioService;
+        this.fileSystem = fileSystem;
+    }
+
+    @PostConstruct
+    protected void init() {
+        try {
+            provisioningRootPath = fileSystem.getPath(PROVISIONING_BRANCH,
+                                                      PROVISIONING_PATH);
+            logger.debug("provisioningRootPath: " + provisioningRootPath.toUri());
+        } catch (Exception e) {
+            //uncommon error
+            logger.error("An error was produced during VFS registries directory initialization.",
+                         e);
+        }
+        entryMarshaller = (VFSRegistryEntryMarshaller) marshallerRegistry.get(VFSRegistryEntry.class);
+    }
+
+    /**
+     * Ensure that an expected directory under the VFS registries host directory exists.
+     * @param directory a directory name.
+     * @return the path to the directory.
+     */
+    public Path ensureDirectory(final String directory) {
+        Path directoryPath = provisioningRootPath.resolve(directory);
+        if (!ioService.exists(directoryPath)) {
+            directoryPath = ioService.createDirectory(directoryPath);
+        }
+        return directoryPath;
+    }
+
+    /**
+     * Helper method calculating the MD5 digest for a String.
+     * @param content a String value for calculating the digest.
+     * @return returns the digest as a 32 character hex string.
+     */
+    public String md5Hex(String content) {
+        if (content == null) {
```

Comment:
In general I don't like to use "values" to represent the semantic of null. I have seen largely people trying to represent null using values, specially in database areas, and then it's very funny to see database registers that has values like
"", "----", "NULL", "null", "nil" and even worst things.
But let's do it, and add a proper comment in the method javadoc.